pages/api/send-analytics.ts

```diff
@@ -0,0 +1,32 @@
+import getConfig from 'next/config';
+import moment from 'moment/moment';
+import {Analytics} from '@segment/analytics-node'
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const { serverRuntimeConfig } = getConfig();
+
+export default async function sendAnalytics(req: NextApiRequest, res: NextApiResponse) {
+    const {event, data = {}} = req?.body;
+    const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
+    const signedUserId = req.cookies['user_id']
+
+    try {
+        const client = new Analytics({
+            writeKey: serverRuntimeConfig.SEGMENT_WRITE_KEY,
+        })
+    
+        client.track({
+            event,
+            anonymousId,
+            properties: {
+                ...data,
+                signedUserId,
+            }
+        })
+    } catch (e) {
+        console.log("Error sending analytics", e)
+    } finally {
+        res.setHeader('Set-Cookie', 'helio_analytics_id=' + anonymousId + '; Path=/; HttpOnly; SameSite=Strict; Max-Age=' + 60 * 60 * 24 * 30)
+        return res.json({success: true})
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnalytics` function is not handling errors properly. It's catching exceptions and logging them, but it's not informing the client about the error. This could lead to silent failures where the client thinks the analytics event was sent successfully when it actually failed. Consider returning an error response in the catch block.

```diff
    } catch (e) {
        console.log("Error sending analytics", e)
+        return res.status(500).json({ success: false, message: 'Failed to send analytics' });
    } finally {
        res.setHeader('Set-Cookie', 'helio_analytics_id=' + anonymousId + '; Path=/; HttpOnly; SameSite=Strict; Max-Age=' + 60 * 60 * 24 * 30)
        return res.json({success: true})
    }
```

Also, consider using a more secure way to generate the `anonymousId`. Using `moment().unix().toString()` might lead to collisions if two requests are made at the same second. You could use a library like `uuid` to generate unique identifiers.

```diff
    const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
+    const anonymousId = req.cookies['helio_analytics_id'] || require('uuid').v4();
```

Lastly, you should await the `client.track` call as it returns a promise. If you don't await it, the function could finish executing and respond to the client before the tracking call has completed.

```diff
        client.track({
+        await client.track({
            event,
            anonymousId,
            properties: {
                ...data,
                signedUserId,
            }
        })
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->